modules/dhcp_libvirt/dhcp_libvirt_main.rb

```diff
@@ -92,5 +92,10 @@ def del_record(_, record)
       logger.error msg = "Error removing DHCP record: #{e}"
       raise Proxy::DHCP::Error, msg
     end
+
+    def find_record(_, record)
```

Comment:
Proxy::DHCP::Libvirt::Provider#find_record returns either a Proxy::DHCP::Reservation or Proxy::DHCP::Lease. You can check if it's the latter in Proxy::DHCP::Libvirt::Provider#del_record instead  -- it skips a call to libvirt and is easier to test (could you add a test please too?).

DHCP provider interface has been recently changed (see https://github.com/theforeman/smart-proxy/blob/develop/modules/dhcp/dhcp_api.rb#L152 and https://github.com/theforeman/smart-proxy/blob/develop/modules/dhcp/dhcp_api.rb#L165). Could you update libvirt provider to support these -- it will need to overload del_records_by_ip and del_records_by_mac base-class implementations?